### ISSUE TITLE
How to diable Meeting registration on joining meeting

### ISSUE THREAD

**Issue by JAGANPS** [decodedhealth/flutter_zoom_plugin#58]
@kevinbayes  @gfiury  @DzungTNgo  @ShimiBaliti  

Hello i am using the plugin and its works  as gem to me but i have problem i am facing the issue when joining the meeting i get a pop up for registration which i don't want user to get registered ,how can i handle this background please suggest me how to implement this i am stuck at this .

Thanks in advance 
Jagan PS

![WhatsApp Image 2021-01-12 at 10 29 15 AM](https://user-images.githubusercontent.com/40148628/104616608-80332380-56b0-11eb-9d4a-edf3a48f29bc.jpeg)
i have attached the screen shot for your references please help me to bypass this problem

**Comment by gfiury** [comment]
Hey @JAGANPS, you shouldn't be facing this problem

```
this.meetingOptions = new ZoomMeetingOptions(
        userId: 'example',
        meetingId: meetingId,
        meetingPassword: meetingPassword,
        disableDialIn: "true",
        disableDrive: "true",
        disableInvite: "true",
        disableShare: "true",
        noAudio: "false",
        noDisconnectAudio: "false"
    );
```
The `userId ` should be a valid Zoom User Id 

The Zoom API to [create an user](https://marketplace.zoom.us/docs/api-reference/zoom-api/users/usercreate)
The Zoom API to [get an user](https://marketplace.zoom.us/docs/api-reference/zoom-api/users/userl)

Let me know if you are already using a valid Id

**Comment by JAGANPS** [comment]
@gfiury  yes I am using a valid zoom id and I am passing the user mail-id as  userId,

![Screenshot 2021-01-16 at 1 35 32 PM](https://user-images.githubusercontent.com/40148628/104806740-b6cb8400-57ff-11eb-9ab1-d07a3d11c40a.png)
when I tested with simulator I am not able to join the meeting but with connection of physical device I am able to connect
![Screenshot 2021-01-16 at 1 37 13 PM](https://user-images.githubusercontent.com/40148628/104806771-f4301180-57ff-11eb-90d6-1383e8d8f521.png)

**Comment by JAGANPS** [comment]
@gfiury @kevinbayes  @DzungTNgo @ShimiBaliti 
 hi when I went through the documentation of zoom I came over the function
void setRegisterWebinarInfo(java.lang.String name,
                            java.lang.String email,
                            boolean cancel)
I am able to set the display name by passing userid but after it I get this function how to handle this at backend 
how to integrate this in my code.

**Comment by gfiury** [comment]
@JAGANPS 

You can't call that from Flutter because that is java code `java.lang.String email`. You need to extend the class [ZoomView.java](https://github.com/decodedhealth/flutter_zoom_plugin/blob/master/android/src/main/java/com/decodedhealth/flutter_zoom_plugin/ZoomView.java)

Here is the Zoom Doc: [InMeetingServiceListener](https://zoom.github.io/zoom-sdk-android/us/zoom/sdk/InMeetingServiceListener.html)

The changes will look like this:
```

import us.zoom.sdk.InMeetingServiceListener;

public class ZoomView  implements ..... InMeetingServiceListener {
...
...
...
     @override
     public void onJoinWebinarNeedUserNameAndEmail(InMeetingEventHandler eventHandler) {
            eventHandler.setRegisterWebinarInfo(this.username, this.email, false);
     }

}
```

The tricky part here is how are you gonna pass the username and the email. This is not a trivial change.

**Comment by JAGANPS** [comment]
> @JAGANPS
> 
> You can't call that from Flutter because that is java code `java.lang.String email`. You need to extend the class [ZoomView.java](https://github.com/decodedhealth/flutter_zoom_plugin/blob/master/android/src/main/java/com/decodedhealth/flutter_zoom_plugin/ZoomView.java)
> 
> Here is the Zoom Doc: [InMeetingServiceListener](https://zoom.github.io/zoom-sdk-android/us/zoom/sdk/InMeetingServiceListener.html)
> 
> The changes will look like this:
> 
> ```
> 
> import us.zoom.sdk.InMeetingServiceListener;
> 
> public class ZoomView  implements ..... InMeetingServiceListener {
> ...
> ...
> ...
>      @override
>      public void onJoinWebinarNeedUserNameAndEmail(InMeetingEventHandler eventHandler) {
>             eventHandler.setRegisterWebinarInfo(this.username, this.email, false);
>      }
> 
> }
> ```
> 
> The tricky part here is how are you gonna pass the username and the email. This is not a trivial change.

**Comment by JAGANPS** [comment]
@kevinbayes @DzungTNgo @ShimiBaliti @gfiury   

error: method does not override or implement a method from a supertype
@Override
^
1 error
When I remove the @override  function still I get the webinar registration 
I still get the same webinar registration which I am not able to pass even when I hard code it I am getting the same issue
![WhatsApp Image 2021-01-12 at 10 29 15 AM](https://user-images.githubusercontent.com/40148628/104616608-80332380-56b0-11eb-9d4a-edf3a48f29bc.jpeg)


public void onJoinWebinarNeedUserNameAndEmail(InMeetingEventHandler eventHandler) {
            eventHandler.setRegisterWebinarInfo("test",""email@gmail.com", false);
     }

I get the same issues please help me to solve this
